core-ui/src/sidebar/useSidebarCondensed.tsx

```diff
@@ -0,0 +1,35 @@
+import { useState, useEffect } from 'react';
+import { useSetRecoilState } from 'recoil';
+import { isSidebarCondensedState } from 'state/preferences/isSidebarCondensedAtom';
+
+function getWindowDimensions() {
+  const { innerWidth: width } = window;
+  return width;
+}
+
+function useWindowDimensions() {
+  const [windowDimensions, setWindowDimensions] = useState(
+    getWindowDimensions(),
+  );
+
+  useEffect(() => {
+    function handleResize() {
+      setWindowDimensions(getWindowDimensions());
+    }
+
+    window.addEventListener('resize', handleResize);
+    return () => window.removeEventListener('resize', handleResize);
+  }, []);
+
+  return windowDimensions;
+}
+
+export default function useSidebarCondensed() {
+  const width = useWindowDimensions();
+
+  const setSidebarCondensed = useSetRecoilState(isSidebarCondensedState);
+
+  if (width <= 900) {
```

Comment:
not sure about mixing block styles. I'd suggest keeping consistent.
Either:
```
if (foo) {
  option1();
} else {
  option2();
}
```
or
```
if (foo)
  option1();
else
  option2();
```
or
```
if (foo) option1();
else option2();
```
but not a mix